test/src/test/kotlin/graphql/nadel/tests/next/fixtures/hydration/virtual/HydrationMakesVirtualFieldTest.kt

```diff
@@ -0,0 +1,164 @@
+                                )
+                            }
+                    }
+            },
+        ),
+        // Service for hydration
+        Service(
+            name = "jira",
+            overallSchema = """
+                type Query {
+                  issuesByIds(ids: [ID!]!): [JiraIssue]
+                }
+                type JiraIssue {
+                  id: ID!
+                  key: String!
+                  title: String!
+                }
+            """.trimIndent(),
+            runtimeWiring = { runtime ->
+                data class JiraIssue(
+                    val id: String,
+                    val key: String,
+                    val title: String,
+                )
+
+                val issuesByIds = listOf(
+                    JiraIssue(
+                        id = "ari:cloud:jira::issue/1",
+                        key = "GQLGW-1",
+                        title = "Fix the speed of light",
+                    ),
+                    JiraIssue(
+                        id = "ari:cloud:jira::issue/2",
+                        key = "GQLGW-2",
+                        title = "Refactor Nadel",
+                    ),
+                ).strictAssociateBy { it.id }
+
+                runtime
+                    .type("Query") { type ->
+                        type
+                            .dataFetcher("issuesByIds") { env ->
+                                val ids = env.getArgument<List<String>>("ids")
+
+                                ids!!
+                                    .map {
+                                        issuesByIds[it]
+                                    }
+                            }
+                    }
+            },
+        ),
+        // Service that introduces virtual type
+        Service(
+            name = "work",
+            overallSchema = """
+                type Query {
+                  business_stub: String @hidden
+                  businessReport_findRecentWorkByTeam(
+                    teamId: ID!
+                    first: Int
+                    after: String
+                  ): WorkConnection
+                    @hydrated(
+                      service: "graph_store",
+                      field: "graphStore_query"
+                      arguments: [
+                        {
+                          name: "query"
+                          value: "SELECT * FROM Work WHERE teamId = ?"
+                        }
+                        {
+                          name: "first"
+                          value: "$argument.first"
+                        }
+                        {
+                          name: "after"
+                          value: "$argument.after"
+                        }
+                      ]
+                    )
+                }
+                directive @virtualType on OBJECT
+                type WorkConnection @virtualType {
+                  edges: [WorkEdge]
```

Comment:
Can we add some scalar s in this test, like `hasNextPage`, `endCursor` etc.

Reply:
Yeah will flesh this out and add some more tests today